src/main/java/com/epam/msu/controller/CertificateController.java

```diff
@@ -0,0 +1,108 @@
+package com.epam.msu.controller;
+
+import com.epam.msu.dto.CertificateDto;
+import com.epam.msu.entity.Response;
+import com.epam.msu.exception.CertificateNotFoundException;
+import com.epam.msu.service.impl.CertificateServiceImpl;
+import com.epam.msu.service.validation.Validation;
+import com.epam.msu.util.ParametersUtils;
+import com.google.protobuf.ServiceException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cglib.core.Local;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+@RestController
+@RequestMapping("/certificates")
+public class CertificateController {
+
+    private final CertificateServiceImpl certificateService;
+
+    @Autowired
+    public CertificateController(CertificateServiceImpl certificateService) {
+        this.certificateService = certificateService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<CertificateDto>> getAllCertificates(@RequestParam(required = false) String tag,
+                                                                   @RequestParam(required = false) String name,
+                                                                   @RequestParam(required = false) String sort) throws CertificateNotFoundException {
+        List<CertificateDto> certificates = certificateService.getAllCertificates();
```

Comment:
All this logic with if checks should be moved to services (not only n this endpoint, but also in others). The good practice is to have only service calls in controller

Reply:
fixed